contracts/wrapper/NameWrapper.sol

```diff
@@ -129,7 +129,19 @@ contract NameWrapper is
 
         bytes32 labelhash = _getEthLabelhash(bytes32(id), fuses);
         if (labelhash != bytes32(0)) {
-            expiry = uint64(registrar.nameExpires(uint256(labelhash))) + GRACE_PERIOD;
+            expiry =
+                uint64(registrar.nameExpires(uint256(labelhash))) +
+                GRACE_PERIOD;
+            // if owner in registrar is not the wrapper, zero out the owner
+            try registrar.ownerOf(uint256(labelhash)) returns (
```

Comment:
You can probably avoid the need for the try/catch if you just check if expiry + grace period is in the future.